arch/risc-v/include/mode.h

```diff
@@ -0,0 +1,91 @@
+/****************************************************************************
+ * arch/risc-v/include/mode.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISC_V_INCLUDE_MODE_H
+#define __ARCH_RISC_V_INCLUDE_MODE_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <arch/csr.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_USE_S_MODE
+
+/* CSR definitions */
+
+#  define CSR_STATUS        sstatus          /* Global status register */
```

Comment:
My thinking was this:
csr.h defines all csr register contents
irq.h defines all irq sources and related stuff
mode.h binds the mode (M- or S-mode) dependent stuff together from csr.h irq.h and whatever will come in the future

What do you think ?

Reply:
Here is my thought:

1. Other arch doesn't have mode.h file
2. Other arch normally define the alternative name in the same file e.g.:
https://github.com/apache/incubator-nuttx/blob/master/arch/arm/include/armv8-m/irq.h#L65-L82

Reply:
Final though on this matter. Other arch's don't have a mode.h file because there is no need for it. They run in either privileged mode or user/privileged mode. For example in ARMthere is PRV_U / PRV_S (or something).

RISC-V is special in this case, as we have to options (modes) for the privileged mode. The kernel can run in either M-mode or S-mode, and this is why I created mode.h!

So in this case, the alternative names are not ubiquitous, they are mode dependent!

Reply:
I can of course do it either way. I was just trying to give a reason why I made the separate file. IMO it is clearer but I can of course move the bindings to the csr/irq files too!